src/datacontainer.js

```diff
@@ -108,27 +115,137 @@ class DataContainer {
+    } catch (error) {
+      // Ignore the 'AuthorizationPermissionMismatch' error that will be throw if the client has read-only rights.
+      // The save of the schema can be done only by the clients with read-write access.
+      if (error.code !== 'AuthorizationPermissionMismatch') {
+        rethrowDebug(`Failed to save the json schema '${this.schema.id}' with error: ${error}`, error);
+      }
+    }
+  }
+
+  /**
+   * If the schema was previously saved, this method will make an integrity check, otherwise will save the schema in
+   * a blockBlob.
+   * @private
+   */
+  async _cacheSchema() {
+    let storedSchema;
+    let schemaName = this._getSchemaName(this.schemaVersion);
+    try {
+      let schemaBlob = await this.blobService.getBlob(this.name, schemaName);
+      storedSchema = schemaBlob.content;
+    } catch (error) {
+      if (error.code === 'BlobNotFound') {
+        this._saveSchema();
+
+        this._validateFunctionMap[this.schemaVersion] = this.validator.compile(this.schema);
+        return;
+      }
+      rethrowDebug(`Failed to save the json schema '${this.schema.id}' with error: ${error}`, error);
+    }
+
+    // integrity check
+    if (storedSchema !== JSON.stringify(this.schema)) {
+      throw new SchemaIntegrityCheckError('The stored schema is not the same with the schema defined.');
+    }
+    this._validateFunctionMap[this.schemaVersion] = this.validator.compile(this.schema);
+  }
+
+  /**
+   * Method that validates the content
+   *
+   * @param content - JSON content
+   * @param schemaVersion - the schema version (optional)
+   *
+   * @return {object}
+   * ```js
+   * {
+   *    valid: boolean,   // true/false if the content is valid or not
+   *    errors: [],       // if the content is invalid, errors will contain an array of validation errors
+   * }
+   * ```
+   */
+  async validate(content, schemaVersion = this.schemaVersion) {
+    let ajvValidate = this._validateFunctionMap[schemaVersion];
+    // if the validate function is not available, this means that the schema is not yet loaded
+    if (!ajvValidate) {
+      // load the schema
+      try {
+        let schemaBlob = await this.blobService.getBlob(this.name, this._getSchemaName(schemaVersion));
+        let schema = JSON.parse(schemaBlob.content);
+        // cache the ajv validate function
+        this._validateFunctionMap[schemaVersion] = this.validator.compile(schema);
+        ajvValidate = this._validateFunctionMap[schemaVersion];
+      } catch (error) {
+        rethrowDebug(`Failed to save the json schema '${this.schema.id}' with error: ${error}`, error);
+      }
+    }
+    let result = {
+      valid: ajvValidate(content),
+      errors: ajvValidate.errors,
+    };
+
+    return result;
+  }
+
+  async init() {
+    // ensure the existence of the data container
+    await this.ensureContainer();
+
+    // cache the JSON schema
+    await this._cacheSchema();
```

Comment:
Thinking about caching of non-default versions, maybe this function could have a default parameter for the schema version and the `validate` function could use that internally to load the validator.  Also, maybe if `_cacheSchema` returned the validator function instead, we could avoid using the `this._validateFunctionMap` outside of the `_cacheSchema` function.